app/lib/katello/resources/candlepin/upstream_consumer.rb

```diff
@@ -42,19 +42,24 @@ def remove_entitlement(entitlement_id)
 
           def get_export(url, client_cert, client_key, ca_file)
```

Comment:
These three methods (get_export, update, post_consumer) are all named inconsistently. Since you're introducing the third, this is probably a good time for a refactor. Is there anything preventing this?

Reply:
They all do have reasons for the naming :stuck_out_tongue: 

If you look vertically it makes a bit more sense:
`Actions::Candlepin::Owner::RegenerateUpstreamIdentityCert` calls
`Katello::Glue::Provider#owner_upstream_regenerate_identity_cert` which calls
`Resources::Candlepin::UpstreamConsumer.post_consumer` which calls
`Katello::Resources::Candlepin::UpstreamCandlepinResource#post`

`Actions::Candlepin::Owner::StartUpstreamExport` calls
`Katello::Glue::Provider#start_owner_upstream_export` which calls
`Resources::Candlepin::UpstreamConsumer.get_export` which calls
`Katello::Resources::Candlepin::UpstreamCandlepinResource#get`

In each of these chains the more "general" name (get, post, etc.) is at the end. These methods are named specifically rather than generally because it made more sense to me to keep that pattern. I'd be open to changing them to something else (also non-general) if that's what you think is needed? But I wouldn't know what to call them. I guess the answer to why they're not named consistently is that they're not part of the same family and don't do the same things.

`get_export` is so named because it's used specifically to start the export of a manifest. `post_consumer` is similar, in that it's used specifically to regenerate the identity cert, and is not a general 'post' method.

 Let me know your thoughts..

Reply:
Thanks for the write up on the rationale and implementing the other changes, I appreciate it.

I'm feeling better about the naming as of now, but I still have some issues. This isn't something you started, but I'm not a fan of the Candlepin endpoint requests that have real generic-looking method parameters and log messages but they expect a certain format of data back. I'd be happier removing the web request type from the method name entirely if these are single-purpose methods.

Maybe `get_export` could be `start_upstream_export` and `post_consumer` could be `regenerate_upstream_identity`? Just some ideas. If I was working with this file and needed another get or post I would have been tempted to use these two methods as generics (and they might have worked); I think changing the names might help.

Reply:
Or alternatively `get_export` and `post_consumer` could be refactored into general-purpose get and post methods. I think the first option is better but I'm happy with either tbh.

Reply:
> Maybe `get_export` could be `start_upstream_export` and `post_consumer` could be `regenerate_upstream_identity`?

sounds good to me! Let's go with that. :+1:

Reply:
I ran into a bump here:

```rb
def retrieve_owner_upstream_export(upstream, zip_file_path, export_id)
        validate_upstream_identity_cert!(upstream)
        data = Resources::Candlepin::UpstreamConsumer.start_upstream_export(
          "#{api_url(upstream)}#{upstream['uuid']}/export/#{export_id}",
          upstream['idCert']['cert'],
          upstream['idCert']['key'], ca_file
        )
        File.write(zip_file_path, data, mode: 'wb')

        true
      end
```

`start_upstream_export` doesn't make sense as a method name here. I'm going to alias the method; see what you think when I push the changes.

Reply:
Do we typically alias stuff like this? I'm not sure about best practices. Everything else looks great, thank you!

Reply:
I guess what I'm asking is would it make more sense to choose a different name in this case?

Reply:
> would it make more sense to choose a different name for `start_upstream_export` in this case?

That's one option, but I can't think of anything besides `start_or_retrieve_upstream_export` which is cumbersome. That, or a more general name, like (wait for it) `get_export`. :smile:  

We do use `alias_method` a few other places in Katello. If we can't think of a better name, I think alias is fine personally.

Reply:
Haha ok I think that sounds great then. Thanks Jeremy!